scripts/prepare

```diff
@@ -6,35 +6,37 @@ set -e
 
 for f in packages/*; do
 	if [[ -f ${f}/package.yaml ]]; then
-		if [[ -z $CHART || $CHART == $(basename -- ${f}) ]]; then
-			url=$(cat ${f}/package.yaml | yq r - url)
-			subdirectory=$(cat ${f}/package.yaml | yq r - subdirectory)
-			type=$(cat ${f}/package.yaml | yq r - type)
-			fields=$(echo ${subdirectory} | awk -F'/' '{print NF}')
-			commit=$(cat ${f}/package.yaml | yq r - commit)
-			if [[ $fields -eq '0' ]]; then
-				fields='1'
-			fi
-			rm -rf ${f}/charts
-			mkdir -p ${f}/charts
-			if [[ $type == 'git' ]]; then
-				mkdir -p /tmp/tmp-charts
-				git clone --depth=1 --no-tags $url /tmp/tmp-charts
-				pwd=$(pwd)
-				cd /tmp/tmp-charts
-				git fetch origin $commit
-				git checkout $commit
-				cd $pwd
+		if [[ (-z $CHART || $CHART == $(basename -- ${f})) ]]; then
+			if [[ $(cat ${f}/package.yaml | yq r - url) ]]; then
+				url=$(cat ${f}/package.yaml | yq r - url)
```

Comment:
Same comment as above; since at this part of the loop we know that `[[ -f ${f}.package.yaml ]]`, let's just bring the `url=` outside / above this if statement, change the if statement to `if [[ -z ${url} ]]`, and do a `continue` if that the `url` is null. That way we don't have to nest all the other stuff below it in the if statement itself.

Reply:
this logic might slightly change once we rebase on https://github.com/rancher/charts/pull/579, but we can revisit this once the rebase is complete.